src/api.ts

```diff
@@ -53,8 +53,11 @@ export const apiFactory = (serviceId: number) => {
 
 	api.post('/api/v2/:worker/clients/', fromLocalHost, (req, res) => {
 		if (!isTrusted(req.body)) {
-			return res.sendStatus(400);
+			return res.status(400).end();
 		}
+		// Immediately respond to minimize time in the client-connect script
+		res.status(200).end();
```

Comment:
This is the biggest win here as it means we immediately respond to the client-connect script which definitely blocks, the other changes are just a general improvement by removing unnecessary body messages and such